HIP/hip-100.md

```diff
@@ -0,0 +1,479 @@
+
+---
+</details>
+
+<details>
+<summary>abstract class Provider</summary>
+
+> abstract class `Provider`
+
+### Methods
+
+##### `getLedgerId` (): `LedgerId`
+
+Return the ID of the current network
+
+---
+
+##### `async getAccountBalance` ( `accountId`: `AccountId` ): `AccountBalance`
+
+Get the balance for an account
+
+---
+
+##### `async getAccountInfo` ( `accountId`: `AccountId` ): `AccountInfo`
+
+Get the info for an account
+
+---
+
+##### `async getTransactionReceipt` ( `transactionId`: `TransactionId` ): `TransacitonReceipt`
+
+Get a receipt for a transaction ID
+
+---
+
+##### `async sendTransaction` ( `transaction`: `Transaction` ): `TransactionResponse`
+
+Sign and send a transaction using the wallet
+
+---
+
+##### `async waitForReceipt` ( `response`: `TransactionResponse` ): `TransactionReceipt`
+
+Wait for the receipt for a transaction response
+
+**NOTE**: This is different than `getTransactionReceipt()` this method requires a `nodeId` which
+is set inside `TransactionResponse` and as a result should not be able to fail with `RECEIPT_NOT_FOUND`
+
+---
+</details>
+
+<details>
+<summary>abstract class Wallet extends Signer</summary>
+
+### Static Methods
+
+##### `withPrivateKey` ( `privateKey`: `PrivateKey` ): `Wallet`
+
+Create a wallet using a private key
+
+---
+
+### Methods
+
+##### `getProvider` (): `Provider`
+
+Return the provider
+
+---
+
+##### `getPublicKey` (): `PublicKey`
+
+Return the public key associated with this wallet.
+
+---
+
+##### `async createRandomED25519` (): `Wallet`
+
+Creates a wallet with a new ED25519 key
+
+**NOTE**: This would create an alias key account on Hedera
+
+---
+
+##### `async createRandomECDSA` (): `Wallet`
+
+Creates a wallet with a new ECDSA key
+
+**NOTE**: This would create an alias key account on Hedera
+
+---
+</details>
+
+<details>
+<summary>class SignerSignature</summary>
+
+### Fields
+
+##### `publicKey`: `PublicKey`
+
+The public key that signed this request
+
+##### `signature`: `bytes`
+
+The siganture for the message
+
+</details>
+
+<details>
+<summary>class LocalWallet implements Wallet</summary>
+
+### Static Methods
+
+### Constructor
+
+##### `constructor`()
+
+Creates an `LocalWallet` from the environment variables `OPERATOR_KEY`, `OPERATOR_ID`, and `HEDERA_NEWTORK`
+
+---
+
+### Methods
+
+##### async `sign` ( `messages`: `List` < `bytes` > ): `List` < `List` < `SignatureProviderSignature` > >
+
+Signs all the messages with all the private keys.
+
+---
+</details>
+
+<details>
+<summary>class VoidSigner extends Signer</summary>
+
+### Static Methods
+
+##### `withAccountId` ( `accountId`: `AccountId` ): `VoidSigner`
+
+Create a wallet using a private key
+
+---
+
+### Methods
+
+##### `getProvider` (): `Provider`
+
+Return the provider
+
+---
+
+##### `getPublicKey` (): `PublicKey`
+
+Will return `null`
+
+---
+
+##### `async signTransaction` ( `transaction`: `Transaction` )
+
+Will do nothing
```

Comment:
Should it not  throw an error as signing a transaction using a `VoidSigner` is not supported.

Reply:
I was a bit conflicted here tbh. One one hand it does make sense for it to throw an error, but on the other handle I feel it also makes sense for it to not do anything. This would allow the SDK to simply use all signers to sign without checking to see if it should attempt to sign with a particular signer.